cpp/src/arrow/compute/api_scalar.h

```diff
@@ -1350,5 +1371,24 @@ ARROW_EXPORT Result<Datum> AssumeTimezone(const Datum& values,
                                           AssumeTimezoneOptions options,
                                           ExecContext* ctx = NULLPTR);
 
+/// \brief Between compares each element in `values`
+/// with `left` as a lower bound and 'right' as an upper bound
+///
+/// \param[in] values input to compare between left and right
+/// \param[in] left used as the lower bound for comparison
+/// \param[in] right used as the upper bound for comparison
+/// \param[in] options for bounds, default is inclusive of both, other
+///  endpoints, other choices are left (exclude left endpoint), right
+///  (exclude right endpoint) and both (exclude both endpoints), optional
+/// \param[in] ctx the function execution context, optional
+///
+/// \return the resulting datum
```

Comment:
Also `\since 7.0.0` (optimistically :slightly_smiling_face:)?

Reply:
Hopefully, may need to update.

Reply:
Updated to 8.0.0